src/site/content/en/blog/storage-foundation/index.md

```diff
@@ -0,0 +1,375 @@
+
+{% include 'content/origin-trials.njk' %}
+
+### Register for the origin trial {: #register-for-ot }
+
+{% include 'content/origin-trial-register.njk' %}
+
+### Feature detection
+
+To check if the Storage Foundation API is supported, use:
+
+```js
+if ('storageFoundation' in window) {
+  // The Storage Foundation API is supported.
+}
+```
+
+### Complete examples
+
+To make the concepts introduced above clearer, here are two complete examples that walk you through
+the different stages in the lifecycle of Storage Foundation files.
+
+#### Opening, writing, reading, closing
+
+```js
+// Open a file (creating it if needed).
+const file = await storageFoundation.open('test_file');
+try {
+  // Request 100 bytes of capacity for this context.
+  await storageFoundation.requestCapacity(100);
+
+  const writeBuffer = new Uint8Array([64, 65, 66]);
+  // Write the buffer at offset 0. After this operation, `result.buffer`
+  // contains the transferred buffer and `result.writtenBytes` is 3,
+  // the number of bytes written. `writeBuffer` is left detached.
+  let result = await file.write(writeBuffer, 0);
+
+  const readBuffer = new Uint8Array(3);
+  // Read at offset 1. `result.buffer` contains the transferred buffer,
+  // `result.readBytes` is 2, the number of bytes read. `readBuffer` is left
+  // detached.
+  result = await file.read(readBuffer, 1);
+  // `Uint8Array(3) [65, 66, 0]`
+  console.log(result.buffer);
+} finally {
+  file.close();
+}
+```
+
+#### Opening, listing, deleting
+
+```js
+// Open three different files (creating them if needed).
+await storageFoundation.open('sunrise');
+await storageFoundation.open('noon');
+await storageFoundation.open('sunset');
+// List all existing files.
+// `["sunset", "sunrise", "noon"]`
+await storageFoundation.getAll();
+// Delete one of the three files.
+await storageFoundation.delete('noon');
+// List all remaining existing files.
+// `["sunrise", "noon"]`
+await storageFoundation.getAll();
+```
+
+## Demo
+
+You can play with the [Storage Foundation API demo][demo] in the embed below. Create, rename, write
+into, and read from files, and see the available capacity you have requested update as you make
+changes. You can find the [source code][demo-source] of the demo on Glitch.
+
+{% Glitch 'storage-foundation' %}
+
+## Security and permissions
+
+The Chromium team has designed and implemented the Storage Foundation API using the core principles
+defined in [Controlling Access to Powerful Web Platform Features][powerful-apis], including user
+control, transparency, and ergonomics.
+
+Following the same pattern as other modern storage APIs on the web, access to the Storage Foundation API is
+origin-bound, meaning that an origin may only access self-created data. It is also limited to secure
+contexts.
+
+### User control
+
+Storage quota will be used to distribute access to disk space and to prevent abuse. Like other
```

Comment:
```suggestion
Storage quota will be used to distribute access to disk space and to prevent abuse. Memory you want to occupy needs to be requested first. Like other
```